internal/error.go

```diff
@@ -769,7 +855,8 @@ func convertFailureToError(failure *failurepb.Failure, dc converter.DataConverte
 	} else if failure.GetTimeoutFailureInfo() != nil {
 		timeoutFailureInfo := failure.GetTimeoutFailureInfo()
 		lastHeartbeatDetails := newEncodedValues(timeoutFailureInfo.GetLastHeartbeatDetails(), dc)
-		err = NewTimeoutError(
+		err = newTimeoutError(
+			failure.GetMessage(),
```

Comment:
`TimeoutFailure` may have message set on server (`WorkflowTask timeout`/`ActivityTask timeout`). This will propagate it to error.